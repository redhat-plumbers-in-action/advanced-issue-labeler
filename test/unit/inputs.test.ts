import { validate } from 'class-validator';
import { describe, it, expect, beforeEach, test } from 'vitest';

import {
  inputsContextFixture,
  IInputsTestContext,
} from './fixtures/inputs.fixture';

describe('Inputs Object', () => {
  beforeEach<IInputsTestContext>(context => {
    context.inputs = inputsContextFixture.inputs;
    context.invalid = inputsContextFixture.invalid;
  });

  it<IInputsTestContext>('can be instantiated', context =>
    context.inputs.map(inputsItem => expect(inputsItem).toBeDefined()));

  test<IInputsTestContext>('get template()', context =>
    context.inputs.map(inputsItem =>
      expect(inputsItem.template).toMatchSnapshot()
    ));

  test.todo('set template()');

  test<IInputsTestContext>('get section()', context =>
    context.inputs.map(inputsItem =>
      expect(inputsItem.section).toMatchSnapshot()
    ));

  test.todo('set section()');

  test<IInputsTestContext>('get blockList()', context =>
    context.inputs.map(inputsItem =>
      expect(inputsItem.blockList).toMatchSnapshot()
    ));

  test.todo('set blockList()');

  it<IInputsTestContext>('is valid', async context =>
    context.inputs.map(async inputsItem =>
      expect(validate(inputsItem)).resolves.toMatchObject([])
    ));

  it<IInputsTestContext>('is invalid', async context =>
    context.invalid.map(async invalidItem =>
      expect(validate(invalidItem)).resolves.toMatchSnapshot()
    ));
});
