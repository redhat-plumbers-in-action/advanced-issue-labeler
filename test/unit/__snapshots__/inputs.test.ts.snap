// Vitest Snapshot v1

exports[`Inputs Object > get blockList() 1`] = `undefined`;

exports[`Inputs Object > get blockList() 2`] = `
[
  "block1",
  "block2",
]
`;

exports[`Inputs Object > get blockList() 3`] = `undefined`;

exports[`Inputs Object > get section() 1`] = `"section"`;

exports[`Inputs Object > get section() 2`] = `"section"`;

exports[`Inputs Object > get section() 3`] = `"section"`;

exports[`Inputs Object > get template() 1`] = `undefined`;

exports[`Inputs Object > get template() 2`] = `undefined`;

exports[`Inputs Object > get template() 3`] = `"template"`;

exports[`Inputs Object > is invalid 1`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "isString": "_section must be a string",
      "minLength": "_section must be longer than or equal to 1 characters",
    },
    "property": "_section",
    "target": Inputs {
      "_blockList": undefined,
      "_section": undefined,
      "_template": undefined,
    },
    "value": undefined,
  },
]
`;

exports[`Inputs Object > is invalid 2`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "isString": "_section must be a string",
      "minLength": "_section must be longer than or equal to 1 characters",
    },
    "property": "_section",
    "target": Inputs {
      "_blockList": undefined,
      "_section": undefined,
      "_template": undefined,
    },
    "value": undefined,
  },
]
`;

exports[`Inputs Object > is invalid 3`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "minLength": "_section must be longer than or equal to 1 characters",
    },
    "property": "_section",
    "target": Inputs {
      "_blockList": undefined,
      "_section": "",
      "_template": undefined,
    },
    "value": "",
  },
]
`;

exports[`Inputs Object > is invalid 4`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "minLength": "each value in _blockList must be longer than or equal to 1 characters",
    },
    "property": "_blockList",
    "target": Inputs {
      "_blockList": [
        "",
      ],
      "_section": "section",
      "_template": undefined,
    },
    "value": [
      "",
    ],
  },
]
`;

exports[`Inputs Object > is invalid 5`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "minLength": "_template must be longer than or equal to 1 characters",
    },
    "property": "_template",
    "target": Inputs {
      "_blockList": [
        "block1",
        "block2",
      ],
      "_section": "section",
      "_template": "",
    },
    "value": "",
  },
]
`;
